tests/e2e/test_query_assets_download_media.py

```diff
@@ -0,0 +1,437 @@
+    """Tests with token protected content url"""
+    with TemporaryDirectory() as tmp_dir:
+        assets = kili.assets(
+            project_id=src_project_image["id"],
+            fields=["externalId", "content", "jsonContent", "project.inputType"],
+        )
+
+        assets = download_assets(assets, tmp_dir)
+
+        assert assets[0]["content"] == os.path.join(str(tmp_dir.resolve()), "kili_logo.png")
+        assert Path(assets[0]["content"]).is_file()
+        assert md5_hash(Path(assets[0]["content"])) == "24814a261785e6f18968d83efa1b6a01"
+        assert assets[0]["jsonContent"] == ""
+        assert assets[0]["externalId"] == "kili_logo"
+
+        assert assets[1]["content"] == os.path.join(str(tmp_dir.resolve()), "kili_logo.png")
+        assert Path(assets[1]["content"]).is_file()
+        assert md5_hash(Path(assets[1]["content"])) == "24814a261785e6f18968d83efa1b6a01"
+        assert assets[1]["jsonContent"] == ""
+        assert assets[1]["externalId"] == "kili_logo.png"
+
+
+@pytest.fixture()
+def src_project_video_frames(kili):
+    """create video project with token protected urls and frames"""
+    interface = {
+        "jobs": {
+            "JOB_0": {
+                "content": {
+                    "categories": {
+                        "OBJECT_A": {"children": [], "name": "Object A"},
+                        "OBJECT_B": {"children": [], "name": "Object B"},
+                    },
+                    "input": "radio",
+                },
+                "instruction": "Categories",
+                "isChild": False,
+                "mlTask": "CLASSIFICATION",
+                "models": {},
+                "isVisible": True,
+                "required": 1,
+            }
+        }
+    }
+
+    project = kili.create_project(
+        input_type="VIDEO",
+        json_interface=interface,
+        title="test_asset.py",
+        description="test_asset.py",
+    )
+
+    with TemporaryDirectory() as tmp_dir:
+        urls = get_json_content_urls_video(
+            "https://storage.googleapis.com/label-public-staging/Frame/vid2_frame/video2_video2-json-content.json"
+        )
+        for i, url in enumerate(urls):
+            if i >= 5:  # keep only a few images for testing
+                break
+            with requests.get(url, stream=True, timeout=20) as response:
+                response.raise_for_status()
+                with open(os.path.join(tmp_dir, f"video2-img{i:06}.jpg"), "wb") as file:
+                    for chunk in response.iter_content(chunk_size=1024 * 1024):
+                        file.write(chunk)
+
+        json_content_array = [
+            [os.path.join(tmp_dir, filename) for filename in sorted(os.listdir(tmp_dir))]
+        ]
+        kili.append_many_to_dataset(
+            project_id=project["id"],
+            content_array=None,
+            external_id_array=["vid 2"],
+            json_content_array=json_content_array,
+        )
+
```

Comment:
add an asset with both content and json_content to simulate that case. (case where you import a video in mp4 format and ask the app to divide it in frames)